catalog/solr/catalog-solr-collection-impl/src/main/java/ddf/catalog/solr/collection/impl/AbstractSolrCollectionProvider.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.collection.impl;
+
+import com.google.common.collect.ImmutableList;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.solr.collection.SolrCollectionProvider;
+import ddf.catalog.util.impl.DescribableImpl;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang.math.NumberUtils;
+import org.codice.solr.factory.SolrCollectionConfiguration;
+import org.codice.solr.factory.SolrConfigurationData;
+
+public abstract class AbstractSolrCollectionProvider extends DescribableImpl
+    implements SolrCollectionProvider {
+  static final List<String> SOLR_CONFIG_FILES =
```

Comment:
I like the critical part of this collection provider API, where we match against a provided metacard. I think it can be reworked to play nicely with the existing provider and deployed with less code. 

❗ However, regarding the configuration features, do they really belong at the Solr client / factory level? For logic that is likely going to be invoked once and then never invoked again? 

I think a lot of these configuration sync tasks should be done separately, not as part of the client. Most of what I'm seeing is just writing local files to Zookeeper. Unless I fundamentally misunderstand how Solr Cloud works, what it expects, or what it needs to function properly, we should remove this capability from the Solr components.

Reply:
yes agree, this class is gonna be clean up significantly after rebasing with this PR https://github.com/codice/ddf/pull/5985